web/src/echo.ts

```diff
@@ -450,13 +449,14 @@ export function reify_message_id(local_id: string, server_id: number): void {
 }
 
 export function update_message_lists({old_id, new_id}: {old_id: number; new_id: number}): void {
-    if (all_messages_data !== undefined) {
-        all_messages_data.change_message_id(old_id, new_id);
-    }
     for (const msg_list of message_lists.all_rendered_message_lists()) {
         msg_list.change_message_id(old_id, new_id);
         msg_list.view.change_message_id(old_id, new_id);
     }
+
+    for (const msg_list_data of message_lists.non_rendered_data()) {
+        msg_list_data.change_message_id(old_id, new_id);
+    }
```

Comment:
Is the thinking behind this sequencing that we want to do the ones that will actually be user-visible first? Makes sense to me, but might be worth writing a comment, since I could imagine someone wanting to combine the loops.

Reply:
Yeah, added comments for this.